src/daemon/ca/manager.rs

```diff
@@ -887,92 +904,236 @@ impl CaManager {
+                            errors.push(Error::CaParentSyncError(
+                                handle.clone(),
+                                parent.clone(),
+                                rcn.clone(),
+                                e.to_string()
+                            ));
+                            break;
+                        },
+                        Ok(response) => {
+                            match response {
+                                rfc6492::Res::Issue(issuance) => {                                    
+                                    // Update the received certificate.
+                                    //
+                                    // In a typical exchange we will only have one key under an RC under a
+                                    // parent. During a key roll there may be multiple keys and requests. It
+                                    // is still fine to update the received certificate for key "A" even if we
+                                    // would get an error for the request for key "B". The reason is such an
+                                    // *unlikely* failure would still trigger an appropriate response at
+                                    // the resource class level in the next loop iteration below.
+                                    let (_, _, _, issued) = issuance.unwrap();
+                                    if let Err(e) = self.send_command(CmdDet::upd_received_cert(
+                                        handle,
+                                        rcn.clone(),
+                                        RcvdCert::from(issued),
+                                        self.config.clone(),
+                                        self.signer.clone(),
+                                        actor,
+                                    ))
+                                    .await {
+                                        // Note that sending the command to update a received certificate
+                                        // cannot fail unless there are bigger issues like this being the wrong
+                                        // response for this resource class. This would be extremely odd because
+                                        // we only just asked the resource class which request to send. Still, in
+                                        // order to handle this the most graceful way we can, we should just drop
+                                        // this resource class and report an error. If there are are still resource
+                                        // entitlements under the parent for this resource class, then a new class
+                                        // will be automatically created when we synchronize the entitlements again.
+
+                                        let reason = format!("received certificate cannot be added, error: {}", e);
+
```

Comment:
Rather than lots of similar/identical code blocks and comments to push an error and break, maybe factor that out?

Reply:
There are subtle differences and if you look at the actual code it's only 2 or 3 calls per option. The remainder is comments. I find it more readable like this despite the duplication.